### ISSUE TITLE
Favicon display issue for Coursera.org

### ISSUE THREAD

**Issue by Arachnopod** [aaFn/Bookmark-search-plus-2#124]
In the normal FF Bookmark sidebar, the www.coursera.org favicon displays as a horizontal blue infinity symbol on a white background.  However, in the Bookmark-search-plus-2 sidebar, the favicon appears to be missing.  If one hovers the mouse over the bookmark item, causing it be highlighted in light blue, you can see a horizontal *white* infinity symbol on a white background.  Don't know why the difference.  Haven't seen this issue with any other of my many, many (>50000) bookmarks.

Of course, the blue symbol on white would be preferable to the appearance of a blank favicon!

**Comment by aaFn** [comment]
Looks like  I am not getting the proper link to the favicon ... I will check in details over the week-end.

`
[Exception... "Favicon at "https://s3.amazonaws.com/favicon.ico" failed to load: Forbidden."  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource:///modules/FaviconLoader.jsm :: onStopRequest :: line 236"  data: no] FaviconLoader.jsm:236:22
`

**Comment by aaFn** [comment]
Found the problem, in fact all is ok, but BSP2 gets the first link to a favicon in the page, which is
https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-194x194.png
and is blank (or nearly .. check the link).

For this page, there are multiple favicon links, and all the others have the blue infinite sign, cf. for ex.
https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-blue-16x16.png

I'll see if it is easy to detect when there are multiple ones, and when it happens, to select the one which has the size closest to 16x16 ..

```
<link rel="icon" type="image/png" href="https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-194x194.png" sizes="194x194">
<link rel="icon" type="image/png" href="https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/android-chrome-192x192.png" sizes="192x192">
<link rel="icon" type="image/png" href="https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-blue-96x96.png" sizes="96x96">
<link rel="icon" type="image/png" href="https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-blue-16x16.png" sizes="16x16">
<link rel="icon" type="image/png" href="https://d3njjcbhbojbot.cloudfront.net/web/images/favicons/icon-blue-32x32.png" sizes="32x32">
```

**Comment by aaFn** [comment]
Resolved with 2.0.72, which is out now, and is handling multiple rel=icon references on a page, by sorting them in increasing size order to get the first one closest to 16x16 which works.
This is actually making favicon fetching a little more robust.

Let me know if ok on your side,

**Comment by aaFn** [comment]
Issue resolved. Thanks; aaFn.
